elastic/apm-server
# apmpackage: copy full field hierarchy

## Motivation/summary

Fix integration package index template mappings by copying the full field hierarchy to the integration package, instead of just flattened leaf fields. Copying only leaf fields means we miss inherited group properties such as `dynamic`.

- set a default `dynamic` property for each data stream mapping
- add `ecs.version` to base fields

## Checklist

~- [ ] Documentation has been updated~

## How to test these changes

1. Install the integration
2. Run apm-server with data streams (e.g. `-E apm-server.data_streams.enabled=true`)
3. Send some HTTP transactions with request headers, check that APM Server indexes them successfully
4. Check that the request headers are not indexed

## Related issues

https://github.com/elastic/apm-server/issues/5707
